src/core/core.controller.js

```diff
@@ -712,23 +718,60 @@ helpers.extend(Chart.prototype, /** @lends Chart */ {
 		me.tooltip.transition(easingValue);
 	},
 
+	/**
+	 * @private
+	 */
+	_getSortedDatasetMetas: function() {
+		var me = this;
+		var datasets = me.data.datasets || [];
+		var result = [];
+		var i, ilen;
+
+		for (i = 0, ilen = datasets.length; i < ilen; ++i) {
+			result.push(me.getDatasetMeta(i));
+		}
+
+		result.sort(compare2Level('order', 'index'));
+
+		return result;
+	},
+
+	/**
+	 * @private
+	 */
+	_getSortedVisibleDatasetMetas: function() {
```

Comment:
Maybe you could have `_getSortedDatasetMetas` take a parameter `filterVisible` and then just have this return `me._getSortedDatasetMetas(true)`

Reply:
done